### PR TITLE
Feature/mr ceo/cart

### PR DESCRIPTION
**Pull request by AllStackDev1**

### CODE REVIEW

**Review by mcbobby7 — APPROVED** [review]

File: src/components/app/actions.js [hunk 1]
```diff
@@ -34,3 +39,14 @@ export const setLoginStatus = () => ({
...
 //call an action to request the user to register
+
+/**
+ * Triggers request to get login status
+ *
+ * @function
+ * @return {void} The {@link actionTypes.SET_CART_COUNT SET_CART_COUNT} action.
+ */
+export const setCartCount = cartCount => ({
+    payload: cartCount,
```
Comment (EngrEric) [inline src/components/app/actions.js:50]:
```
export const setCartCount = payload=> ({
    payload,
```

File: src/components/app/components/Navbar.js [hunk 2]
```diff
@@ -43,7 +56,24 @@ class Navbar extends React.Component {
...
+        if (nextProps.cartCount !== prevState.cartCount) {
+            return {
+                cartCount: nextProps.cartCount,
+            };
+        }
+        return null;
+    }
+
+    componentDidMount() {
+        setInterval(() => {
```
Comment (EngrEric) [inline src/components/app/components/Navbar.js:69]: why are you using `setInterval()` here?

File: src/components/app/components/Navbar.js [hunk 3]
```diff
@@ -43,7 +56,24 @@ class Navbar extends React.Component {
...
+            const newCount = Number(JSON.parse(window.localStorage.getItem('cart')).length || 0);
+            this.props.setCartCount(newCount);
+        }, 100);
+    }
+
     render() {
+        console.log(this.props);
```
Comment (EngrEric) [inline src/components/app/components/Navbar.js:76]: remove `console.log`

File: src/components/app/components/styles.css [hunk 4]
```diff
@@ -1,74 +1,77 @@
...
\ No newline at end of file
+  color: white;
+  background-color: #106021;
+  float: right;
+  margin-right: 100px;
+  margin-top: 40px;
+}
+
+.cart-count-nav > .bg-transparent {
+  background-color: transparent !important;
```
Comment (EngrEric) [inline src/components/app/components/styles.css:76]: it is not recommended to use `!important` , if there is a better way please remove it

File: src/components/app/selectors.js [hunk 5]
```diff
@@ -18,3 +18,13 @@ export const getLoginStatus = state => state[NAME].isLoggedIn;
...
  */
 export const getUserData = state => state[NAME].user;
+
+/**
+ * Selects the <tt>user</tt> key.
+ *
+ * @function
+ * @param {Object} state - redux store state
+ * @return {Number} the state data of the signin which contains user data
+ * {@link module:signin/constants::INITIAL_STATE constants::INITIAL_STATE}).
```
Comment (EngrEric) [inline src/components/app/selectors.js:28]: edit comments to reflect what you are doing

File: src/components/cart/components/Cart.js [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+        quantity: 5,
+    },
+    {
+        id: '3',
+        image: '/rice.jpg',
+        price: 32,
+        product: 'Banana',
+        quantity: 1,
+    },
+];
```
Comment (EngrEric) [inline src/components/cart/components/Cart.js]: move data to constant

File: src/components/cart/components/Cart.js [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+                <div className="cart-top">
+                    <div className="cart-heading">
+                        <h2>
+                            {`Cart`}
+                            <Badge
+                                count={this.state.cartCount}
+                                title={`${this.state.cartCount} item${
+                                    this.state.cartCount > 1 ? 's' : ''
+                                }`}
+                                style={{ backgroundColor: '#001529' }}
```
Comment (EngrEric) [inline src/components/cart/components/Cart.js:113]: no inline styling.
move to `styles.css`

**Review by EngrEric — CHANGES_REQUESTED** [review]
Looks good, just require some changes. and also validate that the feature you added is working as required.

File: src/components/app/components/Navbar.js [hunk 8]
```diff
@@ -98,23 +128,39 @@ class Navbar extends React.Component {
                         </Item>
                     </Menu>
                 )}
+                <Menu
+                    key="cart"
+                    className={`${LEFT_NAV_MENU} cart-count-nav`}
+                    mode={NAV_MODE}
+                    theme={NAV_THEME}
+                >
+                    <Item key="cart" className={'bg-transparent'}>
```
Comment (enyeInc) [inline src/components/app/components/Navbar.js:137]: `className='bg-transparent'`

File: src/components/cart/components/Cart.js [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+    constructor(props) {
+        super(props);
+        this.state = {
+            cart: localStorage.getItem('cart') ? JSON.parse(localStorage.getItem('cart')) : [],
+            cartCount: 0,
+            totalAmount: sessionStorage.getItem('cartTotal')
+                ? sessionStorage.getItem('cartTotal')
+                : 0,
+        };
+    }
```
Comment (enyeInc) [inline src/components/cart/components/Cart.js:20]: why are you storing items on the `localStorage` and `sessionStorage`?

File: src/components/cart/components/Cart.js [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+
+    componentDidMount() {
+        let total = 0;
+        const newCart = [...data];
+        newCart.some(entry => {
+            total += entry.price * entry.quantity;
+        });
+        window.localStorage.setItem('cart', JSON.stringify(newCart));
+        window.sessionStorage.setItem('cartTotal', total);
```
Comment (enyeInc) [inline src/components/cart/components/Cart.js:29]: again why do this object live on the `localStorage` and `sessionStorage`

File: src/components/cart/components/Cart.js [hunk 11]
```diff
@@ -0,0 +1,235 @@
...
+    }
+
+    handleRemoveRow(row) {
+        let total = 0;
+        const newCart = [...this.state.cart];
+        const index = newCart.findIndex(item => row.id === item.id);
+        newCart.splice(index, 1);
+        newCart.some(entry => {
+            total += entry.price * entry.quantity;
+        });
```
Comment (enyeInc) [inline src/components/cart/components/Cart.js:47]: use `Array.reduce` to do the logic from `line 61 - 65`

**Comment by enyeInc** [comment]
@MrCEO63 and @EngrEric Why are you guys storing app level data outside of the redux store? This is an anti pattern and should be fixed right away

File: db/config.js [hunk 12]
```diff
@@ -3,4 +3,5 @@
  *
  * @constant
  */
-export const MONGO_DB_URL = 'mongodb://eric:123abc@ds113358.mlab.com:13358/agroeasydb';
+// export const MONGO_DB_URL = 'mongodb://eric:123abc@ds113358.mlab.com:13358/agroeasydb';
+export const MONGO_DB_URL = 'mongodb://localhost:27017/agroeasydb';
```
Comment (EngrEric) [inline db/config.js:7]: this is local db and may not work for everybody. uncomment the remote part

**Review by EngrEric — COMMENTED** [review]
I think the backend looks good to me. Can you push your frontend changes and mention me the part you are having issues to get the user data.
Also did you test out this backend with postman to make sure its working as desired

**Pull request closed without merging**